source/common/upstream/load_balancer_impl.cc

```diff
@@ -107,6 +127,29 @@ void LoadBalancerBase::recalculatePerPriorityState(uint32_t priority,
   }
 }
 
+// Method iterates through priority levels and turns on/off panic mode.
+void LoadBalancerBase::recalculatePerPriorityPanic() {
+  per_priority_panic_.resize(priority_set_.hostSetsPerPriority().size());
+
+  const uint32_t normalized_total_health = std::min<uint32_t>(
```

Comment:
It seems we are doing this calculation in two places now. Can we make a shared helper with good comments?

Reply:
Done. Added new method to calculate normalized total health.